proxy/src/waiters.rs

```diff
@@ -12,10 +14,10 @@ impl<T> Default for Waiters<T> {
 
 impl<T> Waiters<T> {
     pub fn register(&self, key: String) -> Waiter<T> {
-        let (tx, rx) = mpsc::channel();
+        let (tx, rx) = mpsc::channel(3);
```

Comment:
Wouldn't [oneshot](https://docs.rs/tokio/latest/tokio/sync/oneshot/index.html) work better here? The rebased version of my branch now uses it.

Reply:
Yes, I just didn't want to do it in this PR, to simplify review